runner/local/simple.go

```diff
@@ -88,47 +79,57 @@ func (r *simpleRunner) StatusAll() ([]runner.ProcessStatus, error) {
 	return statuses, nil
 }
 
-func (r *simpleRunner) Abort(run runner.RunId) (runner.ProcessStatus, error) {
-	return r.updateStatus(runner.AbortStatus(run))
+func (r *simpleRunner) Abort(runId runner.RunId) (runner.ProcessStatus, error) {
+	return r.updateStatus(runner.AbortStatus(runId))
 }
 
-func (r *simpleRunner) Erase(run runner.RunId) error {
+func (r *simpleRunner) Erase(runId runner.RunId) error {
 	// Best effort is fine here.
 	r.mu.Lock()
 	defer r.mu.Unlock()
-	if result, ok := r.runs[run]; ok && result.State.IsDone() {
-		delete(r.runs, run)
+	if result, ok := r.runs[runId]; ok && result.State.IsDone() {
+		delete(r.runs, runId)
 	}
 	return nil
 }
 
-// TODO(dbentley): when we timeout or abort, we should include the previous stdout/stderr URIs
-func (r *simpleRunner) updateStatus(new runner.ProcessStatus) (runner.ProcessStatus, error) {
+func (r *simpleRunner) updateStatus(newStatus runner.ProcessStatus) (runner.ProcessStatus, error) {
 	r.mu.Lock()
 	defer r.mu.Unlock()
-	old, ok := r.runs[new.RunId]
+
+	oldStatus, ok := r.runs[newStatus.RunId]
 	if !ok {
-		return runner.ProcessStatus{}, fmt.Errorf("cannot find run %v", new.RunId)
+		return runner.ProcessStatus{}, fmt.Errorf("cannot find run %v", newStatus.RunId)
 	}
-	if old.State.IsDone() {
-		return old, nil
+
+	if oldStatus.State.IsDone() {
+		return oldStatus, nil
 	}
-	r.runs[new.RunId] = new
-	if new.State.IsDone() {
+
+	if newStatus.State.IsDone() {
+		if oldStatus.StdoutRef != "" && newStatus.StdoutRef == "" {
```

Comment:
You can skip the oldStatus.StdoutRef != ""... same result, but it's less code...